include/ginkgo/core/matrix/csr.hpp

```diff
@@ -1042,11 +1079,50 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
         this->make_srow();
     }
 
+    /**
+     * Creates a CSR matrix from already allocated (and initialized) row
+     * pointer, column index and value arrays.
+     *
+     * @note This is the same as the previous constructor but with a default
+     *       strategy.
+     */
+    template <typename ValuesArray, typename ColIdxsArray,
+              typename RowPtrsArray>
+    Csr(std::shared_ptr<const Executor> exec, const dim<2>& size,
+        ValuesArray&& values, ColIdxsArray&& col_idxs, RowPtrsArray&& row_ptrs)
+        : EnableLinOp<Csr>(exec, size),
+          values_{exec, std::forward<ValuesArray>(values)},
+          col_idxs_{exec, std::forward<ColIdxsArray>(col_idxs)},
+          row_ptrs_{exec, std::forward<RowPtrsArray>(row_ptrs)}
+    {
+        GKO_ASSERT_EQ(values_.get_num_elems(), col_idxs_.get_num_elems());
+        GKO_ASSERT_EQ(this->get_size()[0] + 1, row_ptrs_.get_num_elems());
+        initialize_default_strategy(row_ptrs_.get_num_elems());
+        this->make_srow();
+    }
```

Comment:
how about this?
You only need the number of nonzeros to call clac_size, but that is only necessary if you initialize the array yourself, otherwise that would be handled by make_srow calling srow_.resize_and_reset
```suggestion
    : Csr{exec, size, std::forward<ValuesArray>(values), std::forward<ColIdxsArray>(col_idxs), std::forward<RowPtrsArray>(row_ptrs), make_default_strategy(exec)) {}
```